src/Http/Paginators/PageRequestPaginator.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Http\Paginators;
+
+use Saloon\Contracts\Connector;
+use Saloon\Contracts\Request;
+use Saloon\Traits\Request\HasPagedPagination;
+
+// TODO 1: Look into serialising the Connector and original Request,
+//           to ensure that we can rebuild the paginator state without storing the entire multiverse.
+
+class PageRequestPaginator extends RequestPaginator
+{
+    use HasPagedPagination;
+
+    /**
+     * @var int
+     */
+    protected readonly int $originalPage;
+
+    /**
+     * @param \Saloon\Contracts\Connector $connector
+     * @param \Saloon\Contracts\Request $originalRequest
+     * @param int|null $limit
+     * @param int $page
+     */
+    public function __construct(
+        Connector $connector,
+        Request $originalRequest,
+        ?int $limit = null,
+        int $page = 1,
+    ) {
+        parent::__construct($connector, $originalRequest, $limit);
+
+        $this->currentPage = $this->originalPage = $page;
+    }
+
+    /**
+     * @return int
+     */
+    public function totalEntries(): int
+    {
+        // Make sure we have a response.
+        if (is_null($this->currentResponse)) {
+            $this->current();
+        }
+
+        return $this->currentResponse->json('json');
+    }
+
+    /**
+     * @return int
+     */
+    public function totalPages(): int
+    {
+        // Make sure we have a response.
+        if (is_null($this->currentResponse)) {
+            $this->current();
+        }
+
+        return $this->currentResponse->json('last_page');
```

Comment:
Same with this one, can we have something like:

```php
return $this->currentResponse->json($this->getLastPageKey());
```

Reply:
I feel when people extend this logic they may forget to add:

```php
if (is_null($this->currentResponse)) {
    $this->current();
}
```

So maybe we recommend people extend something like: 

```php
protected function resolveTotalPages();
```

Key-word there, "resolve" - like we have everywhere else in Saloon, then you could do something like:

```php
public function totalPages(): int
{
    // Make sure we have a response.
    if (is_null($this->currentResponse)) {
        $this->current();
    }

    return $this->resolveTotalPages() ?? $this->currentResponse->json('last_page');
}
```